utils/digital_interrupts.go

```diff
@@ -11,15 +12,53 @@ import (
 	"go.viam.com/rdk/resource"
 )
 
-// DigitalInterruptConfig describes the configuration of digital interrupt for the board.
-type DigitalInterruptConfig struct {
-	Name string `json:"name"`
-	Pin  string `json:"pin"`
-	Type string `json:"type,omitempty"` // e.g. basic, servo
+// PinConfig describes the configuration of a pin for the board.
+type PinConfig struct {
+	Name       string  `json:"name"`
+	Pin        string  `json:"pin"`
+	Type       PinType `json:"type,omitempty"`        // e.g. gpio, interrupt
+	DebounceMS int     `json:"debounce_ms,omitempty"` // only used with interrupts
+	PullState  Pull    `json:"pull,omitempty"`
+}
+
+// PinType defines the pin types we support.
+type PinType string
+
+// PinGPIO represents GPIO pins.
+const PinGPIO PinType = "gpio"
+
+// PinInterrupt represents interrupt pins.
+const PinInterrupt PinType = "interrupt"
+
+// Pull defines the pins pull state(pull up vs pull down).
+type Pull string
+
+const (
+	// PullUp is for pull ups.
+	PullUp Pull = "up"
+	// PullDown is for pull downs.
+	PullDown Pull = "down"
+	// PullNone is for no pulls.
+	PullNone Pull = "none"
+	// PullDefault is for if no pull was set.
+	PullDefault Pull = ""
+)
+
+// Validate validates that the pull is a valid message.
+func (pull Pull) Validate() error {
+	switch pull {
+	case PullDefault:
+	case PullUp:
+	case PullDown:
+	case PullNone:
+	default:
+		return fmt.Errorf("invalid pull configuration %v, supported pull config attributes are up, down, and none", pull)
+	}
+	return nil
 }
 
 // Validate ensures all parts of the config are valid.
-func (config *DigitalInterruptConfig) Validate(path string) error {
+func (config *PinConfig) Validate(path string) error {
```

Comment:
Are you repeating the validation check from the board's Validation function?

Reply:
Uhhh this is the same validation that board did before, but applied to gpio pins as well as digital interrupts.

Reply:
Can you point me to where this Validate is called, I'm trying to figure out if it's now just a delete job, since we're not longer configuring DigitalInterrupts but configuring Pins.

I am expecting a lot of this code to have to be deleted because of unneeded/no longer relevant functionality, so this may be an instance of that.

Reply:
I actually forgot to call the validate function in board.go, so I have [updated the code](https://github.com/viam-modules/raspberry-pi/blob/062a3e7b94f4c66607b146076d351880d309f061/rpi/board.go#L74-L78) accordingly. Additionally added the pull validation to the function as well since there isn't a reason to do it separately